### PR TITLE
Cassandra 18439 5.0

### PR DESCRIPTION
**Pull request by ekaterinadimitrova2**
Thanks for sending a pull request! Here are some tips if you're new here:
 
 * Ensure you have added or run the [appropriate tests](https://cassandra.apache.org/_/development/testing.html) for your PR.
 * Be sure to keep the PR description updated to reflect all changes.
 * Write your PR title to summarize what this PR proposes.
 * If possible, provide a concise example to reproduce the issue for a faster review.
 * Read our [contributor guidelines](https://cassandra.apache.org/_/development/index.html)
 * If you're making a documentation change, see our [guide to documentation contribution](https://cassandra.apache.org/_/development/documentation.html)
 
Commit messages should follow the following format:

```
<One sentence description, usually Jira title or CHANGES.txt summary>

<Optional lengthier description (context on patch)>

patch by <Authors>; reviewed by <Reviewers> for CASSANDRA-#####

Co-authored-by: Name1 <email1>
Co-authored-by: Name2 <email2>

```

The [Cassandra Jira](https://issues.apache.org/jira/projects/CASSANDRA/issues/)

### CODE REVIEW

File: build.xml [hunk 1]
```diff
@@ -245,27 +245,22 @@
         <string>--add-exports java.base/java.lang.ref=ALL-UNNAMED</string>
         <string>--add-exports jdk.unsupported/sun.misc=ALL-UNNAMED</string>
         <string>--add-exports jdk.compiler/com.sun.tools.javac.file=ALL-UNNAMED</string>
+        <string>--add-exports jdk.attach/sun.tools.attach=ALL-UNNAMED</string>
```
Comment (blerer) [inline build.xml:246]: Why do we need this one? It is apparently not needed by Chronicle Queue?

File: build.xml [hunk 2]
```diff
@@ -245,27 +245,22 @@
...
         <string>--add-opens java.base/jdk.internal.loader=ALL-UNNAMED</string>
         <string>--add-opens java.base/jdk.internal.ref=ALL-UNNAMED</string>
         <string>--add-opens java.base/jdk.internal.reflect=ALL-UNNAMED</string>
         <string>--add-opens java.base/jdk.internal.math=ALL-UNNAMED</string>
         <string>--add-opens java.base/jdk.internal.module=ALL-UNNAMED</string>
         <string>--add-opens java.base/jdk.internal.util.jar=ALL-UNNAMED</string>
         <string>--add-opens jdk.management/com.sun.management.internal=ALL-UNNAMED</string>
-        <string>--add-opens jdk.management/com.sun.management.internal=ALL-UNNAMED</string>
-
         <string>--add-opens java.base/sun.nio.ch=ALL-UNNAMED</string>
```
Comment (blerer) [inline build.xml:256]: If we have an `--add-opens` for `java.base/sun.nio.ch` do we need the `--add-exports` ?
Same question for` java.base/java.lang`

**Review by blerer — COMMENTED** [review]
Should we try to document why the add-opens/export-opens were needed?

File: build.xml (hunk 1 above)
Comment (ekaterinadimitrova2) [reply build.xml:246]: This one was added for the SJK integration with nodetool as part of CASSANDRA-18343, but I added it only to the files and not here back then.

File: build.xml (hunk 2 above)
Comment (ekaterinadimitrova2) [reply build.xml:256]: I see only `--add-opens java.base/java.lang=ALL-UNNAMED`. There is `--add-exports java.base/java.lang.ref=ALL-UNNAMED`, but that one is different.
Valid point for` java.base/sun.nio.ch`. My understanding is yes, but I do not find an affirmative java doc on that so I might prefer the redundancy. The add-exports was added back in C* 4.0. WDYT?

**Comment by ekaterinadimitrova2** [comment]
I added one ticket and the chronicle link, but overall, things shift and get outdated quickly, so I am not sure whether it is a good idea. I guess the best way for people to be sure of what they see is to look at the original ticket where those are added and test. Wondering whether to remove even the chronicle link. WDYT?

File: build.xml (hunk 2 above)
Comment (blerer) [reply build.xml:256]: Could we try to remove it?

**Comment by blerer** [comment]
We should probably either add it for all or remove it for all. Let's remove the chronicle link.

**Comment by ekaterinadimitrova2** [comment]
Feedback addressed, started CI here: https://app.circleci.com/pipelines/github/ekaterinadimitrova2/cassandra/2491/workflows/132da5f4-2b10-4464-b3fc-ba1431e70957

EDIT: one known failure - CASSANDRA-17798

**Review by blerer — APPROVED** [review]

**Comment by ekaterinadimitrova2** [comment]
Committed

**Pull request closed without merging**